print-eap.c

```diff
@@ -148,56 +148,36 @@ static const struct tok eap_aka_subtype_values[] = {
  */
 void
 eap_print(netdissect_options *ndo,
-          const u_char *cp,
+          const u_char *tptr,
           u_int length)
 {
-    const struct eap_frame_t *eap;
-    const u_char *tptr;
-    u_int eap_type, tlen, type, subtype;
-    int count=0, len;
-
-    ndo->ndo_protocol = "eap";
-    tptr = cp;
-    tlen = length;
-    eap = (const struct eap_frame_t *)cp;
-    ND_TCHECK_SIZE(eap);
-    eap_type = GET_U_1(eap->type);
-
-    ND_PRINT("%s (%u) v%u, len %u",
-           tok2str(eap_frame_type_values, "unknown", eap_type),
-           eap_type,
-           GET_U_1(eap->version),
-           GET_BE_U_2(eap->length));
-
-    if (ndo->ndo_vflag < 1)
-        return;
-
-    tptr += sizeof(struct eap_frame_t);
-    tlen -= sizeof(struct eap_frame_t);
-
-    switch (eap_type) {
-    case EAP_FRAME_TYPE_PACKET:
-        ND_TCHECK_1(tptr);
-        type = GET_U_1(tptr);
-        ND_TCHECK_2(tptr + 2);
-        len = GET_BE_U_2(tptr + 2);
-        ND_PRINT(", %s (%u), id %u, len %u",
-               tok2str(eap_code_values, "unknown", type),
-               type,
-               GET_U_1((tptr + 1)),
-               len);
-
-        ND_TCHECK_LEN(tptr, len);
-
-        if (type == EAP_REQUEST || type == EAP_RESPONSE) {
-            /* RFC 3748 Section 4.1 */
-            ND_TCHECK_1(tptr + 4);
-            subtype = GET_U_1(tptr + 4);
-            ND_PRINT("\n\t\t Type %s (%u)",
-                   tok2str(eap_type_values, "unknown", subtype),
-                   subtype);
-
-            switch (subtype) {
+    u_int type, subtype;
+    int len, count;
```

Comment:
Should `len` be unsigned?
(line 183: `len = GET_BE_U_2(tptr + 2);`)

Reply:
The eap_print() API had the "len" attribute as signed previously and hence, didnot want to disturb it and retained the same.  

But, as you pointed out, it should have been unsigned in the first place. Hence, changed it to unsigned now. Thanks.